tree/treeplayer/src/TCsvDS.cxx

```diff
@@ -0,0 +1,255 @@
+#include <ROOT/TCsvDS.hxx>
+#include <ROOT/TDFUtils.hxx>
+#include <ROOT/TSeq.hxx>
+
+#include <algorithm>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+namespace ROOT {
+namespace Experimental {
+namespace TDF {
+
+// Regular expressions for type inference
+TRegexp TCsvDS::intRegex("^[-+]?[0-9]+$");
+TRegexp TCsvDS::doubleRegex1("^[-+]?[0-9]+\\.[0-9]*$");
+TRegexp TCsvDS::doubleRegex2("^[-+]?[0-9]*\\.[0-9]+$");
+TRegexp TCsvDS::trueRegex("^true$");
+TRegexp TCsvDS::falseRegex("^false$");
```

Comment:
Do these regexes deserve unit tests for themselves only? @amadio, can you also have a look? You have quite some experience with TRegexp and stl regex.

Reply:
What is the context where these expressions will be used? A `^` usually marks beginning of a line, but in TRegexp this is probably going to be different, right? For `double`, one regex should be enough (the second one, to accomodate `.1`, etc). We still need to update it to be able to parse scientific notation numbers, such as `1.0E-3` for example. The current expression will fail for that. We may also want/need expressions for hexadecimals, octal, etc, to cover what `sscanf()` can read, more or less. We may also want to support `True`/`TRUE`, `False`/`FALSE`, and `0`/`1` for booleans.

Reply:
Ah, there are also infinity and NaNs to think about...

Reply:
In first approximation we can leave hex and oct unsupported I think :sweat_smile: 

The `^` and `$` will mark beginning and end of the field here as the field is extracted from the line before it is matched with the regex.

I think we need both regexes for the doubles, the second one does not match `1.`.

Reply:
Yes, the two double regexes are necessary because TRegexp does not support regexes with unions. The union is therefore implemented as a C++ or :)
The regexes could indeed be extended to support `1.0E-3` as well.

Reply:
Regarding the booleans, one thing we can do (which is actually what pandas does) is to support any combination of uppercase/lowercase in true and false (e.g. True, False, TRUE, FALSE but also TrUe and falsE). For that, we can just convert to lowercase and compare to `true` and `false`.

0/1 for booleans is trickier because they would naturally be classified as integers.

Reply:
If 0/1 are written, we treat them as integers, for TDF it will not be a problem to cast that later if needed. For the upper and lowercase, we can go for a *tolower* if that is not too difficult.